docs/csharp/roslyn-sdk/get-started/syntax-analysis.md

```diff
@@ -0,0 +1,222 @@
+The tree has 1 elements in it.
+The tree has 4 using statements. They are:
+        System
+        System.Collections
+        System.Linq
+        System.Text
+The first member is a NamespaceDeclaration
+There are 1 members declared in this namespace.
+The first member is a ClassDeclaration
+There are 1 members declared in the Program class
+The first member is a MethodDeclaration
+The return type of the Main method is void
+The method has 1 parameters
+The type of the args parameter is string[]
+The body text of the Main method follows:
+        {
+            Console.WriteLine("Hello, World!");
+        }
+```
+
+### Query methods
+
+In addition to traversing trees, you can also explore the syntax tree using the query methods defined on <xref:Microsoft.CodeAnalysis.SyntaxNode?displayProperty=nameWithType>. These methods should be immediately familiar to anyone familiar with XPath. You can use these methods with LINQ to quickly find things in a tree. The <xref:Microsoft.CodeAnalysis.SyntaxNode?displayProperty=nameWithType> has query methods such as <xref:Microsoft.CodeAnalysis.SyntaxNode.DescendantNodes%2A>, <xref:Microsoft.CodeAnalysis.SyntaxNode.AncestorsAndSelf%2A> and <xref:Microsoft.CodeAnalysis.SyntaxNode.ChildNodes>.
+
+You can use these query methods to find the argument to the `Main` method as an alternative to navigating the tree. Add the following code to the bottom of your `Main` method:
+
+[!code-csharp[Query the tree for the arguments to Main](../../../../samples/csharp/roslyn-sdk/SyntaxQuickStart/HelloSyntaxTree/Program.cs#8 "Query the tree for the arguments to Main")]
+
+The first statement uses a LINQ expression and the <xref:Microsoft.CodeAnalysis.SyntaxNode.DescendantNodes%2A> method to locate the same parameter as in the previous example.
+
+Run the program, and you can see that the LINQ expression found the same parameter as manually navigating the tree.
+
+The sample uses `WriteLine` statements to display information about the syntax trees as they are traversed. You can also learn much more by running the finished program under the debugger. You can examine more of the properties and methods that are part of the syntax tree created for the hello world program.
+
+## Syntax walkers
+
+Often you want to find all nodes of a specific type in a syntax tree, for example, every property declaration in a file. By extending the <xref:Microsoft.CodeAnalysis.CSharp.CSharpSyntaxWalker?displayProperty=nameWithType> class and overriding the <xref:Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor.VisitPropertyDeclaration(Microsoft.CodeAnalysis.CSharp.Syntax.PropertyDeclarationSyntax)> method, you process every property declaration in a syntax tree without knowing its structure beforehand. <xref:Microsoft.CodeAnalysis.CSharp.CSharpSyntaxWalker?displayProperty=nameWithType> is a specific kind of <xref:Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor?displayProperty=nameWithType> that recursively visits a node and each of its children.
```

Comment:
nit: do you really want the full name in all the instances?

![image](https://user-images.githubusercontent.com/12971179/36687897-b2e15e6a-1adf-11e8-9b65-94dc9261c64b.png)

Reply:
I looked at the entire doc and use `nameWithType` on the first use, but only the name after that.